### ISSUE TITLE
[Gallery apps Accessibility]: improve window transition semantics for accessibility services on Android

### ISSUE THREAD

**Issue by vick08** [material-components/material-components-flutter-gallery#517]
## Symptom
When moving between the screens inside Gallery apps, such as Shrine or Rally, TalkBack does not place the focus on the first item on the screen.

## Steps to reproduce
* Install and launch Flutter Gallery app.
* Start TalkBack, Settings > Accessibility > TalkBack.
* Open either Shrine or Rally examples.
* Click around on various screens, including the "back" button and observe spoken feedback.

## Expected results
As the screens change, Talkback should (possibly) announce the title of the new fragment / activity and/or place the focus on the first item on the screen.

## Possible cause
TalkBack does not receive accessibility window events that signal appearance or disappearance of windows or fragments. If TalkBack is not aware that windows have changed, no focus management strategy can be applied and no window title changes can be spoken.

CC/ @goderbauer

**Comment by goderbauer** [comment]
Do you know which event TalkBack needs to properly set the focus? As far as I can tell from the code we are sending TYPE_WINDOW_CONTENT_CHANGED whenever any semantics in the current view have changed and TYPE_WINDOW_STATE_CHANGED when we detect a route change. Are we missing an event that we should be sending?
